src/test/pages/AdminUserManager.test.tsx

```diff
@@ -0,0 +1,444 @@
+      name: /find by user role:/i,
+    }),
+  postcode: () =>
+    screen.getByRole("textbox", {
+      name: /find users with school within a given distance of postcode:/i,
+    }),
+  postcodeRadius: () => getById("postcode-radius-search"),
+  schoolURN: () => screen.getByRole("textbox", { name: /find a user with school URN/i }),
+  searchButton: () => screen.getByRole("button", { name: "Search" }),
+};
+
+const mockSuccessfulPostRequest = (route: string) => {
+  return rest.post(API_PATH + route, (req, res, ctx) => {
+    return res(ctx.status(200), ctx.json({}));
+  });
+};
+
+describe("Admin User Manager", () => {
+  const renderUserManager = async ({ role = "ADMIN" }: { role?: UserRole } = {}) => {
+    renderTestEnvironment({
+      role: role,
+      PageComponent: AdminUserManager,
+      initialRouteEntries: ["/admin/usermanager"],
+      extraEndpoints: [
+        rest.get(API_PATH + "/admin/users", (req, res, ctx) => {
+          const mockAdminSearchResults = [
+            mockUser,
+            { ...buildMockStudent(101), emailVerificationStatus: "NOT_VERIFIED" },
+            buildMockTeacher(102),
+          ];
+          return res(ctx.status(200), ctx.json(mockAdminSearchResults));
+        }),
+        rest.get(API_PATH + "/users/school_lookup", (req, res, ctx) => {
+          const mockSchoolLookup = {
+            "1": { name: "Test School" },
+            "101": { name: "N/A" },
+            "102": { name: "N/A" },
+          };
+          return res(ctx.status(200), ctx.json(mockSchoolLookup));
+        }),
+        rest.delete(API_PATH + "/admin/users/:userId", (req, res, ctx) => {
+          return res(ctx.status(204), ctx.json({}));
+        }),
+        ...[
+          "/users/resetpassword",
+          "/admin/users/change_role/:role",
+          "/admin/users/change_email_verification_status/:status/true",
+          "/admin/users/teacher_pending/:status",
+          "/admin/users/merge",
+        ].map((route) => mockSuccessfulPostRequest(route)),
+      ],
+    });
+    await waitFor(() => {
+      const userState = store.getState().user;
+      expect(userState?.loggedIn).toBe(true);
+    });
+  };
+
+  describe("Search section", () => {
+    it("renders with expected page title and has all expected fields", async () => {
+      await renderUserManager();
+      checkPageTitle("User manager");
+      const { searchForm, familyName, email, schoolOther, role, postcode, postcodeRadius, schoolURN, searchButton } =
+        searchFields;
+      [
+        searchForm(),
+        familyName(),
+        email(),
+        schoolOther(),
+        role(),
+        postcode(),
+        postcodeRadius(),
+        schoolURN(),
+        searchButton(),
+      ].forEach((input) => expect(input).toBeInTheDocument());
+      const expectedRadiusOptions = ["5 miles", "10 miles", "15 miles", "20 miles", "25 miles", "50 miles"];
+      const radiusOptions = within(postcodeRadius()).getAllByRole("option");
```

Comment:
When you write this type of test, it would be a bit safer to put an assertion here for the length of `radiusOptions`, because if at some point that were to return an empty array, this wouldn't fail, which is not right. The same goes for `roleOptions`. Does this make sense, though?

Reply:
Happy to add in a length check. However the test would would fail on the next assertion `radiusOptions.forEach((option, index) => {
        expect(option).toHaveTextContent(expectedRadiusOptions[index]);
      });` if the array were empty though surely? Same situation for `roleOptions` since I explicitly check that each option has a specific value.

Reply:
No, it wouldn't fail, that's why I suggested checking for the expected length. If `radiusOptions` was an empty array, then the contents of that `forEach` will never be executed. But it wouldn't fail either. You can test it yourself: manually set `radiusOptions` to an empty array (something like `const radiusOptions: HTMLElement[] = [];`) and run the tests. They will pass.

Reply:
😫 I totally misunderstood how that would behave. Added a length check to both!

Reply:
Thank you! This is yet another reason why I like writing tests that are as simple as possible.